### PR TITLE
Static analysis fixes

### PR DESCRIPTION
**Pull request by SimY4**
Findbugs static analysis fixes

### CODE REVIEW

File: awaitility/src/main/java/org/awaitility/reflect/FieldTypeMatcherStrategy.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public boolean matches(Field field) {
 
     /** {@inheritDoc} */
     @Override
-    public void notFound(Class<?> type, boolean isInstanceField) throws FieldNotFoundException {
+    public Field notFound(Class<?> type, boolean isInstanceField) throws FieldNotFoundException {
```
Comment (szpak) [inline awaitility/src/main/java/org/awaitility/reflect/FieldTypeMatcherStrategy.java]: Hmm, what would you like to achieve keeping `void` in the interface?

File: awaitility/src/test/java/org/awaitility/AwaitilityTest.java [hunk 2]
```diff
@@ -475,11 +475,11 @@ public Integer call() throws Exception {
     private abstract class AssertExceptionThrownInAnotherThreadButNeverCaughtByAnyThreadTest {
         public AssertExceptionThrownInAnotherThreadButNeverCaughtByAnyThreadTest() throws Exception {
             ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-            System.setErr(new PrintStream(byteArrayOutputStream, true));
+            System.setErr(new PrintStream(byteArrayOutputStream, true, "UTF-8"));
```
Comment (szpak) [inline awaitility/src/test/java/org/awaitility/AwaitilityTest.java]: I missed it before, but I would use [`StandardCharsets.UTF_8`](https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8) here.

Comment (szpak) [inline awaitility/src/test/java/org/awaitility/AwaitilityTest.java]: x2

Comment (SimY4) [inline awaitility/src/test/java/org/awaitility/AwaitilityTest.java]: StandardCharsets are only available in Java 7 or higher.

Comment (szpak) [inline awaitility/src/test/java/org/awaitility/AwaitilityTest.java]: You are right. I completely forgot that Awaitility support Java 6+.

**Comment by johanhaleby** [comment]
Thanks for the PR

**Pull request merged**